nmdc_server/auth.py

```diff
@@ -22,74 +25,190 @@
     "401": {"description": "Login required"},
     "403": {"description": "Insufficient permissions"},
 }
-oauth2_client = starlette_client.OAuth(starlette_config)
+oauth2_client = starlette_client.OAuth()
 oauth2_client.register(
     name="orcid",
-    client_id=settings.client_id,
-    client_secret=settings.client_secret,
-    server_metadata_url=settings.open_id_config_url,
+    client_id=settings.orcid_client_id,
+    client_secret=settings.orcid_client_secret,
+    server_metadata_url=settings.orcid_openid_config_url,
     client_kwargs={
-        "scope": settings.oauth_scope,
+        "scope": settings.orcid_authorize_scope,
     },
 )
 
-oauth2_scheme = OAuth2AuthorizationCodeBearer(
-    authorizationUrl=settings.oauth_authorization_endpoint,
-    tokenUrl=settings.oauth_token_endpoint,
-    scopes={"authorization": settings.oauth_scope},
-    auto_error=False,
+bearer_scheme = HTTPBearer(auto_error=False)
+
+router = APIRouter()
+
+CREDENTIAL_DECODE_EXCEPTION = HTTPException(
+    status_code=status.HTTP_401_UNAUTHORIZED,
+    detail="Could not validate credentials",
+    headers={"WWW-Authenticate": "Bearer"},
+)
+
+CREDENTIAL_MISSING_EXCEPTION = HTTPException(
+    status_code=status.HTTP_401_UNAUTHORIZED,
+    detail="Login required",
+    headers={"WWW-Authenticate": "Bearer"},
 )
 
+AUTHORIZATION_CODE_INVALID_EXCEPTION = HTTPException(
+    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+    detail="Invalid authorization code",
+)
 
-class Token(BaseModel):
-    access_token: UUID
-    token_type: str
-    refresh_token: UUID
-    expires_in: int
-    scope: str
-    name: str
-    orcid: str
-    expires_at: int
+API_JWT_ALGORITHM = "HS256"
+API_JWT_ISSUER = "https://data.microbiomedata.org"
```

Comment:
I think I waffled on that, and now I can't remember why I did it that way. Let me revisit that.

Reply:
Okay I've updated things so that the `host` field from `Settings` is used as the issuer. This means that the `host` _must_ be set now. I've given it the default value of `http://127.0.0.1:8000` and we already override that value on dev, prod, and sandbox, so I think that should be safe.